src/mod_muc_light_db_odbc.erl

```diff
@@ -502,5 +502,9 @@ update_room_version_transaction({RoomU, RoomS}, Version) ->
 main_host({_, RoomS}) ->
```

Comment:
this function should be more simple:
- accept only MucServer or US;
- don't make assumptions about host type (i.e. MucServer should be always MucServer, not Server sometimes).

Reply:
I agree, the main cause not that `main_host/1` doesn't handle the case when it's given a `Server` instead of `MucServer`; the main cause is that a function calling `main_host/1` gets a  `Server` when it doesn't expect it.

Reply:
Yes, of course it is the cause.

Reply:
And this is the thing that should be fixed